ctfdpy/ctfdpy/ctfd.py

```diff
@@ -9,3 +13,39 @@ def __init__(self, url: str, token: str) -> None:
 
     def _generate_packages(self):
         self.users = Users(self.http)
+        self.teams = Teams(self.http)
+
+    def __enter__(self):
+        return self
+
+    def __exit__(self, exc_type, exc_value, traceback):
+        self.http.client.close()
+        return False
+
+
+    def create_full_team(self, name: str, password: str, *members: UserInput, **kwargs: dict[str, Any]) -> None:
+        """
+        Generate a complete team with the given parameters and its members already attached
+
+        Args:
+            name (str): The name of the team
+            password (str): The password of the team
+            members (UserInput): A tuple of the users to attach to the team
+            kwargs (dict[str, Any]): Additional parameters to pass to the team
+
+        Raises:
+            CreationError: The team could not be created
+
+        Returns:
+            _type_: The team created with all its informations
+        """
+        team = self.teams.create_team(name, password)
+        if not team:
+            raise CreationError("Team could not be created")
+
+        for member in members:
+            new_member = self.users.create_user(member.name, member.email, member.password)
+            if not new_member:
+                continue # TODO : Should we raise an error here?
```

Comment:
Should we raise an CreationError here or just continue but it can be confusing for the user and knowing `HTTPClient` already raises some errors ?
- Try/Except the errors and continue anyway to avoid blocking the creation ad mid-point (having half-full team)
- Raise a CreationError knows the state of the team (but needs to do a `get_team`)

Reply:
What do you think @Anemys ?